core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -249,7 +249,7 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
       error("Driver memory must be a positive number")
     }
     if (executorMemory != null
-        && Try(JavaUtils.byteStringAsBytes(executorMemory)).getOrElse(-1L) <= 0) {
+        && Try(JavaUtils.byteStringAsMb(executorMemory)).getOrElse(-1L) <= 0) {
```

Comment:
Since this change `executorMemory`, do we need to update line 248 for `driverMemory` together maybe?

Reply:
BTW, do we need to change this line? This seems to ensure that the value is non-negative. Just for my understanding, could you give me some example which gives different result before and after this PR?

Reply:
You are right, there is no difference regarding the behaviour but reading the code and seeing `byteStringAsBytes` called with these configs gives the false impression they are in bytes. I think it is worth to change them to `byteStringAsMb`.